src/wasm.cc

```diff
@@ -596,6 +604,14 @@ std::shared_ptr<PluginHandleBase> getOrCreateThreadLocalPlugin(
   }
   auto plugin_handle = plugin_factory(wasm_handle, plugin);
   local_plugins[key] = plugin_handle;
+  wasm_handle->wasm()->wasm_vm()->addFailCallback([key](proxy_wasm::FailState fail_state) {
+    if (fail_state == proxy_wasm::FailState::RuntimeError) {
+      // If VM failed, erase the entry so that:
+      // 1) we can recreate the new thread local plugin from the same base_wasm.
+      // 2) we wouldn't reuse the failed VM for new plugin configs accidentally.
+      local_plugins.erase(key);
```

Comment:
I don't think that's enough. What about `local_wasms`? If we don't immediately remove that entry, then we might reuse the crashed WasmVM upon plugin restart, notably, when plugin doesn't immediately shutdown (e.g. returns `false` in `on_done` callback).

Reply:
deletion in local_wasms is already done in the callback set in the above (`getOrCreateThreadLocalWasm`) for the vm_key used here. So I relies on that.. am I missing something here?